File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {

        ListNode dummy = new ListNode(-1);
        ListNode current = dummy;

        ListNode cur1 = list1;
        ListNode cur2 = list2;

        while(cur1!=null && cur2!=null){
            if(cur1.val<=cur2.val){
                ListNode temp = cur1;
                cur1 = cur1.next;
                temp.next = null;
                current.next = temp;
            }else{
                ListNode temp = cur2;
                cur2 = cur2.next;
                temp.next = null;
                current.next = temp;
            }
            current = current.next;
        }

        if(cur1!=null){
            current.next = cur1;
        }

        if(cur2!=null){
            current.next = cur2;
        }

        return dummy.next;
        
    }
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:

        dummy = ListNode(-1)
        current = dummy

        cur1 = list1
        cur2 = list2

        while cur1 and cur2:
            if cur1.val <= cur2.val:
                temp = cur1
                cur1 = cur1.next
                temp.next = None
                current.next = temp
            else:
                temp = cur2
                cur2 = cur2.next
                temp.next = None
                current.next = temp
            current = current.next

        if cur1:
            current.next = cur1
        
        if cur2:
            current.next = cur2

        return dummy.next
        